state/etcdstatedriver.go

```diff
@@ -211,7 +215,11 @@ func (d *EtcdStateDriver) ReadAllState(baseKey string, sType core.State,
 	return ReadAllStateCommon(d, baseKey, sType, unmarshal)
 }
 
-func (d *EtcdStateDriver) channelStateEvents(sType core.State,
+// ChannelStateEvents watches for updates(created, modify, delete) to a state of
+// specified type and unmarshals (given a function) all changes and puts then on
+// channel of core.WatchState objects.
+// XXX: move this to some common file
+func ChannelStateEvents(d core.StateDriver, sType core.State,
```

Comment:
> Is this a new public function exposed for the state driver now, which was embedded into WatchAll earlier

No, this is an existing function. It is now shared by etcd and consul state-driver. Actually, thinking more it need not be public as both the drivers are in same `state` package. I will make this function private.

> but I saw your comment about why Watchall can miss some events due to timing. Please clarify.

This function itself is not related to WatchAll missing events. Regarding the comment itself, I realized that condition during my implementation of consul statedriver. Today the way we use watch (which is only when the netplugin instance starts), we first restore all the state (by calling ReadAll()) and then start the vent loop (by calling WatchAll()). This introduces a small time window (between call to ReadAll() and WatchAll()) where any changes in state might get missed.

This is not a major issue today because we use watch to monitor configuration state, which usually is not changing that quickly. Never the less I wanted to note it down so that we are aware of it and can address it if needed.